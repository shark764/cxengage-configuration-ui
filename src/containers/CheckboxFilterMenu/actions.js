import * as ACTIONS from './constants';

export function toggleMenuItems(itemName, menuType) {
  return {
    type: ACTIONS.TOGGLE_MENUITEMS,
    itemName,
    menuType
  };
}
export function updateGroupsData(arrayOfGroupData) {
  return {
    type: ACTIONS.SET_GROUPS_DATA,
    arrayOfGroupData
  };
}
export function updateSkillsData(arrayOfSkillData) {
  return {
    type: ACTIONS.SET_SKILLS_DATA,
    arrayOfSkillData
  };
}
export function setVisibleMenu(menuType) {
  return {
    type: ACTIONS.SET_VISIBLE_MENU,
    menuType
  };
}
export function toggleAllMenuItemsOn(menuType) {
  return {
    type: ACTIONS.TOGGLE_ALL_MENUITEMS_ON,
    menuType
  };
}
export function toggleAllMenuItemsOff(menuType) {
  return {
    type: ACTIONS.TOGGLE_ALL_MENUITEMS_OFF,
    menuType
  };
}

export function toggleAllInverseMenuItems(menuType) {
  return {
    type: ACTIONS.TOGGLE_INVERSE_MENUITEMS,
    menuType
  };
}

export function toggleTimeFormat() {
  return {
    type: ACTIONS.TOGGLE_TIME_FORMAT
  };
}
