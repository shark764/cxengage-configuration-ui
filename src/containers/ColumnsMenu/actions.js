import * as ACTIONS from './constants';

export function toggleColumn(columnName) {
  return {
    type: ACTIONS.TOGGLE_COLUMN,
    columnName
  };
}

export function toggleAllColumnsOn() {
  return {
    type: ACTIONS.TOGGLE_ALL_COLUMNS_ON
  };
}
export function toggleAllColumnsOff() {
  return {
    type: ACTIONS.TOGGLE_ALL_COLUMNS_OFF
  };
}

export function toggleAllInverseColumns() {
  return {
    type: ACTIONS.TOGGLE_INVERSE_COLUMNS
  };
}

export function toggleTimeFormat() {
  return {
    type: ACTIONS.TOGGLE_TIME_FORMAT
  };
}
