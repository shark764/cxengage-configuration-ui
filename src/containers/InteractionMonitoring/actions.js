import * as ACTIONS from './constants';

export function updateTableData(arrayOfTableData) {
  return {
    type: ACTIONS.SET_TABLE_DATA,
    arrayOfTableData
  };
}
export function setExpanded(expanded) {
  return {
    type: ACTIONS.SET_EXPANDED,
    expanded
  };
}
export function setFiltered(filtered) {
  return {
    type: ACTIONS.SET_FILTERED,
    filtered
  };
}
export function setSelected(selected, expanded) {
  return {
    type: ACTIONS.SET_SELECTED,
    selected,
    expanded
  };
}
export function setSorted(sorted) {
  return {
    type: ACTIONS.SET_SORTED,
    sorted
  };
}
export function removeSelected() {
  return {
    type: ACTIONS.REMOVE_SELECTED
  };
}
