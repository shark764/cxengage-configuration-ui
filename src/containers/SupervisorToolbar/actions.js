import * as ACTIONS from './constants';

export function setTwilioEnabled() {
  return {
    type: ACTIONS.SET_TWILIO_ENABLED
  };
}
