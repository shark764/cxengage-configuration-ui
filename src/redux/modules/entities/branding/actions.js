/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

export const fetchBranding = () => ({
  type: 'FETCH_BRANDING_$'
});
