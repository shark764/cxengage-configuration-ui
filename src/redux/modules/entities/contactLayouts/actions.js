/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

export const removeContactLayoutsListItem = (type, deleteEntityId) => {
  if (type === 'contactLayoutsListItem') {
    return {
      type: 'REMOVE_CONTACT_LAYOUTS_LIST_ITEM',
      contactLayoutsListItemId: deleteEntityId
    };
  } else if (type === 'categoryItems') {
    return {
      type: 'REMOVE_CONTACT_LAYOUTS_LIST_ITEM',
      categoryId: deleteEntityId
    };
  }
};
