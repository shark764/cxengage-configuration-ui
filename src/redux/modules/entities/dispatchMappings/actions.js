/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

/*
 *
 * DispatchMappings ACTIONS
 *
 */

export const setMappingValueInvalid = (invalid = false) => ({
  type: 'SET_MAPPING_VALUE_INVALID',
  invalid
});
