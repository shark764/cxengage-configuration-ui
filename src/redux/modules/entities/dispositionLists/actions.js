/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

/*
 *
 * DispositionLists ACTIONS
 *
 */

export const removeDispositionListItem = (type, deleteEntityId) => {
  if (type === 'dispositionListItem') {
    return {
      type: 'REMOVE_DISPOSITION_LIST_ITEM',
      dispositionListItemId: deleteEntityId
    };
  } else if (type === 'categoryItems') {
    return {
      type: 'REMOVE_DISPOSITION_LIST_ITEM',
      categoryId: deleteEntityId
    };
  }
};
