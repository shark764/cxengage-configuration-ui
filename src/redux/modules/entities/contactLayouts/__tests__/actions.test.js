/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import { removeContactLayoutsListItem } from '../actions';

describe('removeContactLayoutsListItem', () => {
  it('when type is contactLayoutsListItem', () => {
    expect(removeContactLayoutsListItem('contactLayoutsListItem', 'mockId')).toMatchSnapshot();
  });
  it('when type is categoryItems', () => {
    expect(removeContactLayoutsListItem('categoryItems', 'mockId')).toMatchSnapshot();
  });
});
