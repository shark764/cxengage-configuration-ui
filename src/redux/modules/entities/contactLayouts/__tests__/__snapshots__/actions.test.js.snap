// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeContactLayoutsListItem when type is categoryItems 1`] = `
Object {
  "categoryId": "mockId",
  "type": "REMOVE_CONTACT_LAYOUTS_LIST_ITEM",
}
`;

exports[`removeContactLayoutsListItem when type is contactLayoutsListItem 1`] = `
Object {
  "contactLayoutsListItemId": "mockId",
  "type": "REMOVE_CONTACT_LAYOUTS_LIST_ITEM",
}
`;
